packages/blade/src/components/Skeleton/_decisions/decisions.md

```diff
@@ -0,0 +1,326 @@
+
+```jsx
+<Skeleton isLoading={data.isLoading}>
+  <Box
+    display="flex"
+    alignItems="center"
+    justifyContent="center"
+    width="200px"
+    height="200px"
+  >
+    {data.title}
+  </Box>
+</Skeleton>
+```
+
+#### Predefined dimensions 
+
+**Typography components:**
+
+Similar to Amount component we can expose a `type` prop which will have predefined sizes & we also expose `numberOfLines` prop which will dictate how many skeletons it will map to.
+
+> numberOfLines prop can also be used to add a trailing effect to the last line of the skeleton
+> Say we have `numberOfLines={5}` then we can render the 5th Skeleton a bit shorter than the others to convey the text is trailing.
+
+```jsx
+<Skeleton 
+  type="
+    | body-small
+    | body-medium
+    | body-large
+    | heading-small
+    | heading-medium
+    | heading-large
+    | title-small
+    | title-medium
+    | title-large
+    | title-xlarge
+  " 
+  numberOfLines={5} 
+/>
+```
+
+A real world usecase might look something like:
+
+<img src="./real-usage-demo-before.png" alt="Card example without skeleton" width="50%" />
+
+```jsx
+<Box
+  padding="spacing.5"
+  display="flex"
+  flexDirection="column"
+  gap="spacing.2"
+  backgroundColor="surface.background.level2.lowContrast"
+>
+  <Box display="flex" flexDirection="column" gap="spacing.3">
+    <Skeleton type="heading-medium" width="70%" borderRadius="max" />
+    <Skeleton type="title-large" width="50%" />
+    <Skeleton type="body-medium" width="70%" />
+  </Box>
+  <Skeleton marginTop="10px" height="50px" borderRadius="4px" />
+  <Divider />
+  <Skeleton type="body-medium" numberOfLines={2} />
+</Box>
+```
+
+<img src="./real-usage-demo-after.png" alt="Card example with Skeleton" width="50%" />
+
+**Block components:**
+
+And for block components we won't give any predefined sizes instead consumers can compose via width, height, borderRadius: 
+
+```jsx
+<Skeleton width="100px" height="100px" borderRadius="medium" />
+```
+
+#### Conclusion
+
+Given the complexity of inferring the dimensions and how it differs from component to component, where for Typography components we need to render the skeleton inside and for block level components we need to wrap it with Skeleton, Plus the complexities that might arise with react-native implementation. It's better to go with predefined dimensions for typography components and for block level components let consumers handle it manually via width/height prop. 
+
+## Motion
+
+Check the motion [here](https://www.figma.com/proto/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?type=design&node-id=16502-258345&scaling=min-zoom&page-id=16498%3A256331):
+
+https://github.com/razorpay/blade/assets/35374649/8298efde-f977-4aa3-bc2d-92fb50a8c830
+
+
+## Accessibility
+
+- All Skeleton nodes will have aria-hidden to hide them from screen readers
+- All Skeleton nodes will have `cursor: progress`
+- Consumers will have to add aria-busy in the respective container which is being rendered
```

Comment:
I'm not sure if this will be followed by consumers. Thoughts about exposing a `SkeletonWrapper` that only adds `aria-busy`? Seems like an overkill but I'm sure a lot of consumers would miss adding aria-busy

Reply:
Yes to option is to expose SkeletonGroup or similar wrapper component to handle it, but we need to make sure to enforce people to use SkeletonGroup. 

But even with SkeletonGroup, it might not be correct, imagine 5 cards with SkeletonGroup component inside each card's body now the page will have 5 aria-busy, but ideally there should be only 1 aria-busy for that whole section.

Reply:
Will explore a11y demos on storybook https://github.com/razorpay/blade/pull/1379#discussion_r1259421017